### ISSUE TITLE
Bug: Add Password Validation Policy to Postgres Module #376

### ISSUE THREAD

**Issue by manojchinnala** [terraform-google-modules/terraform-google-sql-db#430]
### TL;DR

In Release v14.0.0 Password Validation Policy to postgres module is released.
It is working fine for existing postgresql instances (primary and read replica) and breaking when we try to create new primary instance and read replica instance together. 
When tried to create primary instance alone it is working fine without any issue. 

### Expected behavior

Password validation policies should work fine, when creating a new primary instance and read replica. 

### Observed behavior

**##Error**

│ Error: Error, failed to create instance pgsql-cftv14-replica-1: googleapi: Error 400: Invalid request: Password Validation policies cannot be set on a read replica., invalid

### Terraform Configuration

```hcl
password_validation_policy_config = {
    # Complexity Default - password must contain at least one lowercase, one uppercase, one number and one non-alphanumeric characters.
    complexity                  = "COMPLEXITY_DEFAULT"
    disallow_username_substring = true
    min_length                  = 8
    # Password change interval format is in seconds. 3600s=1h
    password_change_interval = "3600s"
    reuse_interval           = 1
  }
```

### Terraform Version

```sh
Terraform v1.3.9
on windows_amd64
+ provider registry.terraform.io/hashicorp/google v4.55.0
+ provider registry.terraform.io/hashicorp/google-beta v4.55.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
```
